carts/src/test/java/com/oracle/coherence/examples/sockshop/micronaut/carts/TracingTest.java

```diff
@@ -0,0 +1,132 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@MicronautTest
+@Property(name = "tracing.jaeger.enabled", value = "true")
+public class TracingTest {
+
+	@Inject
+	EmbeddedServer server;
+
+	@Inject
+	TracerFactory.SafeInMemoryReporter reporter;
+
+	@BeforeEach
+	void setup() {
+		RestAssured.reset();
+		RestAssured.baseURI = "http://localhost";
+	}
+
+	@Test
+	void testTracing() {
+		TestBody testBody = () -> {
+			given().
+					port(server.getPort()).
+					contentType(ContentType.JSON).
+					body(new Item("X1", 0, 10f)).
+				when().
+					post("/carts/{cartId}/items", "C1").
+				then().
+					statusCode(HttpStatus.CREATED.getCode()).
+					body("itemId", Is.is("X1"),
+							"quantity", Is.is(1),
+							"unitPrice", Is.is(10f));
+
+			Blocking.sleep(250);
+		};
+		runTest(testBody);
+
+
+		JaegerSpan[] localSpans = validateOpsPresent(
+				new String[]{"POST /carts/{customerId}/items", "Invoke.process"},
+				reporter.getSpans()
+		);
+		Arrays.stream(localSpans).forEach(TracingTest::validateTagsForSpan);
+	}
+
+	private void runTest(TestBody testBody) {
+		try {
+			testBody.run();
+		} catch (Exception e) {
+			throw Base.ensureRuntimeException(e);
+		}
+	}
+
+	protected static JaegerSpan[] validateOpsPresent(String[] sOpNames, List<JaegerSpan> spans) {
+		assertThat(sOpNames, is(notNullValue()));
+		assertThat("Spans not recorded.", spans.isEmpty(), is(false));
+
+		JaegerSpan[] spansFound = new JaegerSpan[sOpNames.length];
+		for (int i = 0, len = sOpNames.length; i < len; i++) {
+			if (spansFound[i] == null) {
+				final String searchFor = sOpNames[i];
+
+				spansFound[i] = spans.stream().filter(
+						span -> Base.equals(searchFor, span.getOperationName())).findFirst().orElse(null);
+			}
+		}
+
+		for (int i = 0, len = sOpNames.length; i < len; i++) {
+			assertThat("Unable to find operation " + sOpNames[i] + " in spans on the member.",
+					spansFound[i],
+					is(notNullValue()));
+		}
+		return spansFound;
+	}
+
+	@SuppressWarnings("SameParameterValue")
+	protected static void validateTagsForSpan(JaegerSpan span) {
+		assertThat(span, is(notNullValue()));
+		Map<String, Object> metadata = span.getTags();
+		String sOpName = span.getOperationName();
+
+		CacheFactory.log("Validating span: " + span + ", tags: " + span.getTags());
+
+		if (sOpName.startsWith("POST")) {
+			assertThat(sOpName + ": incorrect http path",
+					metadata.get("http.path"),
+					is("/carts/C1/items"));
+		}
+
+		if (sOpName.startsWith("Invoke")) {
+			assertThat(sOpName + ": incorrect cache name",
+					metadata.get("cache"),
+					is("carts"));
+		}
+	}
+
+	@FunctionalInterface
+	protected interface TestBody {
```

Comment:
I think we can remove this functional interface.  It had a specific usage in Coherence testing, but that usage doesn't apply here.

Reply:
This would apply of course to other tests using it.

Reply:
Fixed.